### PR TITLE
feat: add contract interaction from app

### PR DESCRIPTION
**Pull request by aon**
# What ❔
- Add the ability to interact with contracts from the block explorer app
- Slightly refactor rpc proxy

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.

### CODE REVIEW

File: packages/private-rpc/src/rpc/rpc-method-handlers.ts [hunk 1]
```diff
@@ -59,50 +45,35 @@ const eth_call: MethodHandler = {
...
     });
 
     if (!isAddressEqual(address, context.currentUser)) {
-      throw new Error('Wrong caller');
+      return unauthorized(id);
     }
 
     if (!tx.to) {
-      throw new Error('no target');
+      return unauthorized(id);
```
Comment (calvogenerico) [inline packages/private-rpc/src/rpc/rpc-method-handlers.ts:76]: I don't believe this is an unhautorized.

File: packages/private-rpc/src/rpc/rpc-method-handlers.ts [hunk 2]
```diff
@@ -59,50 +45,35 @@ const eth_call: MethodHandler = {
...
+  async handle(context, method, params, id) {
     const rawTx = hexSchema.parse(params[0]);
     const tx = parseTransaction(rawTx);
     const address = await recoverTransactionAddress({
       serializedTransaction: rawTx as any,
     });
 
     if (!isAddressEqual(address, context.currentUser)) {
-      throw new Error('Wrong caller');
+      return unauthorized(id);
```
Comment (calvogenerico) [inline packages/private-rpc/src/rpc/rpc-method-handlers.ts:72]: do you believe that having the same error for this case than for when you actually don't have permission is fine? I feel that it's a little confusing for a final user.

File: packages/app/src/composables/useContext.ts [hunk 3]
```diff
@@ -69,7 +70,7 @@ export default (): Context => {
 
   function getL2Provider() {
     if (!l2Provider) {
-      l2Provider = new Provider(currentNetwork.value.rpcUrl);
+      l2Provider = new Provider(rpcUrl.value ?? "");
```
Comment (calvogenerico) [inline packages/app/src/composables/useContext.ts:73]: Should we remove all the other networks? I feel that we should not allow the user to choose the other networks like sepolia or mainnet.

**Review by calvogenerico — APPROVED** [review]

File: packages/app/src/composables/useContext.ts (hunk 3 above)
Comment (aon) [reply packages/app/src/composables/useContext.ts:73]: This is hardcoded in production I believe. We should look into how it's done and modify it accordingly.

**Pull request merged**